Mathlib/CategoryTheory/Limits/Shapes/Reflexive.lean

```diff
@@ -173,6 +194,365 @@ instance (priority := 100) hasCoreflexiveEqualizers_of_hasEqualizers [HasEqualiz
+  | left : WalkingReflexivePairHom one zero
+  | right : WalkingReflexivePairHom one zero
+  | reflexion : WalkingReflexivePairHom zero one
+  | left_reflexion : WalkingReflexivePairHom one one
+  | right_reflexion : WalkingReflexivePairHom one one
+  | id (X : WalkingReflexivePair) : WalkingReflexivePairHom X X
+  deriving DecidableEq
+
+attribute [-simp, nolint simpNF] WalkingReflexivePairHom.id.sizeOf_spec
+
+open WalkingReflexivePairHom
+
+/-- Composition of morphisms in the diagram indexing reflexive (co)equalizers -/
+def WalkingReflexivePairHom.comp :
+    ∀ { X Y Z : WalkingReflexivePair } (_ : WalkingReflexivePairHom X Y)
+      (_ : WalkingReflexivePairHom Y Z), WalkingReflexivePairHom X Z
+  | _, _, _, id _, h => h
+  | _, _, _, h, id _ => h
+  | _, _, _, reflexion, left => id zero
+  | _, _, _, reflexion, right => id zero
+  | _, _, _, reflexion, right_reflexion => reflexion
+  | _, _, _, reflexion, left_reflexion => reflexion
+  | _, _, _, left, reflexion => left_reflexion
+  | _, _, _, right, reflexion => right_reflexion
+  | _, _, _, right_reflexion, right_reflexion => right_reflexion
+  | _, _, _, right_reflexion, left_reflexion => right_reflexion
+  | _, _, _, right_reflexion, right => right
+  | _, _, _, right_reflexion, left => right
+  | _, _, _, left_reflexion, left => left
+  | _, _, _, left_reflexion, right => left
+  | _, _, _, left_reflexion, right_reflexion => left_reflexion
+  | _, _, _, left_reflexion, left_reflexion => left_reflexion
+
+theorem WalkingReflexivePairHom.id_comp
+    {X Y : WalkingReflexivePair} (g : WalkingReflexivePairHom X Y) : comp (id X) g = g := by
+  cases g <;> rfl
+
+theorem WalkingReflexivePairHom.comp_id
+    {X Y : WalkingReflexivePair} (f : WalkingReflexivePairHom X Y) : comp f (id Y) = f := by
+  cases f <;> rfl
+
+theorem WalkingReflexivePairHom.assoc {X Y Z W : WalkingReflexivePair}
+    (f : WalkingReflexivePairHom X Y) (g: WalkingReflexivePairHom Y Z)
+    (h : WalkingReflexivePairHom Z W) : comp (comp f g) h = comp f (comp g h) := by
+  cases f <;> cases g <;> cases h <;> rfl
+
+instance walkingReflexivePairCategory : SmallCategory WalkingReflexivePair where
+  Hom := WalkingReflexivePairHom
+  id := id
+  comp := comp
+  comp_id := comp_id
+  id_comp := id_comp
+  assoc := assoc
+
+@[simp]
+theorem walkingReflexivePairHom_id (X : WalkingReflexivePair) :
+    WalkingReflexivePairHom.id X = 𝟙 X := by rfl
+
+@[simp]
+theorem walkingReflexivePairHom_left_comp_reflexion : left.comp reflexion = left_reflexion := rfl
+
+@[simp]
+theorem walkingReflexivePairHom_right_comp_reflexion : right.comp reflexion = right_reflexion := rfl
+
+@[simp]
+theorem WalkingReflexivePairHom.id.sizeOf_spec' (X : WalkingReflexivePair) :
+    (WalkingReflexivePairHom._sizeOf_inst X X).sizeOf (𝟙 X) = 1 + sizeOf X := by cases X <;> rfl
+
+/-- The forgetful functor forgetting the common section -/
+def forgetReflexion : WalkingParallelPair ⥤ WalkingReflexivePair :=
```

Comment:
A better name should be found because this is not a forgetful functor. Depending on the chosen name, this may have to be put in the `WalkingReflexivePair` or `WalkingParallelPair` namespace.